tests_mozilla.py

```diff
@@ -583,40 +583,49 @@ def test_get_sld_Mixed_case3(self):
         assert 'example.com' == publicsuffix.get_sld('WwW.example.COM')
 
     def test_get_sld_Leading_dot1(self):
-        assert 'com' == publicsuffix.get_sld('.com')
+        # '.com' -> <empty>.<com> -> None, empty labels are not allowed
+        assert None == publicsuffix.get_sld('.com')
 
     def test_get_sld_Leading_dot2(self):
-        assert 'example' == publicsuffix.get_sld('.example')
+        # '.example' -> <empty>.<example> -> None, empty labels are not allowed
+        assert None == publicsuffix.get_sld('.example')
 
     def test_get_sld_Leading_dot3(self):
         assert 'example.com' == publicsuffix.get_sld('.example.com')
 
     def test_get_sld_Leading_dot4(self):
-        assert 'example' == publicsuffix.get_sld('.example.example')
+        # non-strict: TLD:example, SLD:example.example
+        assert 'example.example' == publicsuffix.get_sld('.example.example')
 
     def test_get_sld_Unlisted_sld1(self):
         assert 'example' == publicsuffix.get_sld('example')
 
     def test_get_sld_Unlisted_sld2(self):
-        assert 'example' == publicsuffix.get_sld('example.example')
+        # non-strict: TLD:example, SLD:example.example
+        assert 'example.example' == publicsuffix.get_sld('example.example')
```

Comment:
this is a change where the original logic was flawed -- didn't follow the algorithm complete-- so the test itself was incorrect